peekaboo/db.py

```diff
@@ -244,6 +246,24 @@ def __init__(self, db_url, instance_id=0,
         self.deadlock_backoff_base = 10
         self.connect_backoff_base = 2000
 
+        # prepare a case statement to order results by enum value, i.e. badness
+        badness_when = {member.name: member.value for member in Result}
+        sorting_column = SampleInfo.result
+
+        # postgres is particular regarding case statements in that it add
+        # explicit type conversion to everything but the sorting colum
+        # specification leading to error:
+        # operator does not exist: result = character varying
+        #
+        # It is unclear at this point whether this is happening at the
+        # sqlalchemy or asyncpg level. Let's just work around it for now by
+        # adding that type conversion ourselves.
+        if backend == 'postgresql':
+            sorting_column = sorting_column.name + "::varchar"
+
+        self.badness_case = sqlalchemy.sql.expression.case(
```

Comment:
So here it certainly paid off to have a fresh go at this with some time inbetween. This made me find https://stackoverflow.com/questions/23614956/specify-the-sort-order-for-an-enum-field-type which was easily adjusted to our needs. So now we can continue to store the text representation of the result, sort by its badness and continue to support sqlite.